### PR TITLE
First try for including Hybrid Edge Deployment demo in to the roles

### PR DESCRIPTION
**Pull request by alezzandro**
##### SUMMARY

As discussed with RHPDS team this is the first try for including Hybrid Edge Deployment demo in to the roles
##### ISSUE TYPE
- New role Pull Request

##### COMPONENT NAME
ocp-workload-edge-deployments

##### ADDITIONAL INFORMATION
*PLEASE NOTE*: the role is actually partial complete. This demo requires an All-In-One standalone OpenShift installation with *also* an additional standalone RHEL system to work with. Actually this pull request only refers to creating the environment needed for the OpenShift part, after that I need to verify with RHPDS team how I can create and configure a RHEL host (outside from OCP), setup all the pre-requisites and deploy the SSH key that I create through this Ansible Role.

### CODE REVIEW

**Comment by fridim** [comment]
hello,

Can you fix the following errors returned by CI:


```
  in "/home/travis/build/redhat-cop/agnosticd/tests/static/../../ansible/roles/ocp-workload-edge-deployments/tasks/workload.yml", line 86, column 11
found unacceptable key (unhashable type: 'dict')
  in "/home/travis/build/redhat-cop/agnosticd/tests/static/../../ansible/roles/ocp-workload-edge-deployments/tasks/workload.yml", line 86, column 12
```

```yaml
- name: Installing Cockpit and Docker (it will be replaced by Podman as soon as Cockpit-Podman will become stable)
  become: true
  yum:
    name: {{ item }}
    state: present
  with_items:
    - docker
    - cockpit
    - cockpit-docker
```

Should be written:

```yaml
- name: Installing Cockpit and Docker (it will be replaced by Podman as soon as Cockpit-Podman will become stable)
  become: true
  yum:
    name:
      - docker
      - cockpit
      - cockpit-docker
```

**Comment by alezzandro** [comment]
@fridim it should be fixed now, I used notation:
```
yum:
    name: ['docker', 'cockpit', 'cockpit-docker']
```

I've also completed the whole role, it should work fine, I've also tested against "OCP Workshop demo lab".

Thanks

File: ansible/roles/ocp-workload-edge-deployments/files/deploy-containers-apb/playbooks/deprovision.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 - name: Adding new target_host to the target_group
   hosts: localhost
-  gather_facts: no
```
Comment (fridim) [inline ansible/roles/ocp-workload-edge-deployments/files/deploy-containers-apb/playbooks/deprovision.yml]: Why would you need to gather_facts here ?

File: ansible/roles/ocp-workload-edge-deployments/tasks/remove_workload.yml [hunk 2]
```diff
@@ -1,6 +1,44 @@
 ---
 # Implement your Workload removal tasks here
+- name: Cleaning up the environment
+  debug:
+    msg: "Cleaning up the environment.."
+
+- name: Delete project IoT Development
+  shell: "oc delete project iot-development"
+  ignore_errors: true
```
Comment (fridim) [inline ansible/roles/ocp-workload-edge-deployments/tasks/remove_workload.yml]: What is this workload target ?
is it a shared cluster or is it a cluster that will be also destroyed ?

If it's a shared-cluster, then please do not use `ignore_errors` here

File: ansible/roles/ocp-workload-edge-deployments/tasks/remove_workload.yml [hunk 3]
```diff
@@ -1,6 +1,44 @@
...
+
+- name: Delete project IoT Testing
+  shell: "oc delete project iot-testing"
+  ignore_errors: true
```
Comment (fridim) [inline ansible/roles/ocp-workload-edge-deployments/tasks/remove_workload.yml]: same

File: ansible/roles/ocp-workload-edge-deployments/tasks/remove_workload.yml [hunk 4]
```diff
@@ -1,6 +1,44 @@
...
+
+- name: Delete project IoT Hub
+  shell: "oc delete project iot-hub"
+  ignore_errors: true
```
Comment (fridim) [inline ansible/roles/ocp-workload-edge-deployments/tasks/remove_workload.yml]: same

File: ansible/roles/ocp-workload-edge-deployments/tasks/remove_workload.yml [hunk 5]
```diff
@@ -1,6 +1,44 @@
...
+
+- name: Delete BuildConfig in openshift namespace
+  shell: "oc delete bc deploy-containers-apb -n openshift"
+  ignore_errors: true
```
Comment (fridim) [inline ansible/roles/ocp-workload-edge-deployments/tasks/remove_workload.yml]: same

File: ansible/roles/ocp-workload-edge-deployments/tasks/remove_workload.yml [hunk 6]
```diff
@@ -1,6 +1,44 @@
...
+
+- name: Delete ImageStream in openshift namespace
+  shell: "oc delete is deploy-containers-apb -n openshift"
+  ignore_errors: true
```
Comment (fridim) [inline ansible/roles/ocp-workload-edge-deployments/tasks/remove_workload.yml]: same

**Comment by fridim** [comment]
Hello @alezzandro 
Something didn't work git-wise.
This PR has too many changes and touches too many files.

You should try to import your changes from scratch and force-update the branch associated with this PR or create a new PR.

**Pull request closed without merging**